docs/Technical-Documentation/tech-memos/cypress-auth.md

```diff
@@ -0,0 +1,91 @@
+# Cypress E2E
+
+**Audience**: TDP Software Engineers <br>
+**Subject**:  Cypress Refactor <br>
+**Date**:     October 16th, 2024 <br>
+
+## Summary
+Digging into our pipeline failures associated in ticket #3141, it was found that our cypress authentication was not persisting for the admin user within a single routine that also used the STT user. Further investigation showed our cypress code is not easily extensible and has not only issues with CSRF compliance but scenario-specific authentication as opposed to abstracted and compartmented sessions. While splitting the scenario and/or using `cypress.wait()` might temporarily solve one problem, we have uncovered technical debt requiring refactoring of this code.
+
+### Background
+Debugging the failures within the pipeline had 3 recurring issues:
+1. Referer not found as in [this post](https://github.com/cypress-io/cypress/issues/16975)
+2. `adminApiRequest` failed to update status, resulting in next step failure.
+3. Errors regarding Django's 'csrf_middleware_token`.
+
+By addressing authentication in a standard way and storing all session cookies and tokens, we should be able to resolve these 3 issues.
+
+## Out of Scope
+* Any changes to frontend ReactJS and Nginx apps
+* Significant changes to backend authentication
+* New Cypress workflows beyond our end-to-end test against deployed develop branch
+
+## Method/Design
+
+### Abstracted Gherkin Steps
+Presently, many of the defined Javascript functions for a given Gherkin step are bespoke or single-use instead of abstracted and should be adapted. Additionally, it was found that sessions were lingering between Gherkin scenarios as we did not have generic `setup` and `teardown` implementations ahead of these. Sufficient utilization of abstraction within the scenarios which are now doing setup/teardown between scenarios and proper session management should result in a cleaner Cypress execution and make future additions simpler.
```

Comment:
Could you provide an example of what level of abstraction your are driving towards?

Reply:
Or is that what the `### Abstracted utility authentication functions` section is covering?

Reply:
The abstracted section below is for the JavaScript side. I will revise this section to be more verbose on how I imagine we might restructure the Gherkin. Reading through it, especially without Jan's clearing of cookies, one scenario needed to be run before another, etc. Assuming we get the setup/teardown, these scenarios can be independent. I'll include psuedo-code examples as we discussed Friday I think on how tech memos ought to look.